### ISSUE TITLE
TensorFlow profiler running into OOM issue on GPU

### ISSUE THREAD

**Issue by rahul-fnu** [tensorflow/tensorflow#61512]
### Issue type

Support

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

binary

### TensorFlow version

2.11.0.5

### Custom code

No

### OS platform and distribution

Linux CentOS 7.9.2009

### Python version

3.10

### Current behavior?

Running TensorFlow profiler for longer than 10 second period crashes the inference process because of OOM error and the profiler returns DEADLINE_EXCEEDED. Is there anyway to limit the sampling rate or way to reduce the amount of information being collected to avoid crashing the process?

### Standalone code to reproduce the issue

```shell
`tensorflow_profiler.experimental.client("grpc://localhost:3222", "profiles", 30000)`
```

**Comment by sushreebarsa** [comment]
@rahul-fnu 
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thank you!

**Comment by rahul-fnu** [comment]
@sushreebarsa This is exactly what I run after importing tensorflow: `tensorflow_profiler.experimental.client("grpc://localhost:3222", "profiles", 30000)`

**Comment by Divin-18** [comment]
Reduce Input Size

**Comment by rahul-fnu** [comment]
Is there way to configure the profiler? Either by reducing the amount of information being collected or by reducing the sampling rate?

**Comment by sushreebarsa** [comment]
@rahul-fnu Please refer to [this](https://www.tensorflow.org/tensorboard/tensorboard_profiling_keras) guide to know the way to configure the TF profiler ? Please make sure to use the latest TF version and TF profiler. 
Thank you!

**Comment by ndeepesh** [comment]
@sushreebarsa @Divin-18 
This profiler is used for debugging production issues. We cant reduce the input size. Is there a sampling parameter which we can use so that not all events are captured?

**Comment by trisolaran** [comment]
because of tracing and profiling, the amount of data accumulated is proportionate to the time of profile.
I would suggest that you profile a small period of time, and check if kernel or host side info is more verbose, this is really more work load specific properties.
if host side info is verbose and not very useful, you can specify host_trace_level = 0 (I am vague on if this is available on 2.11)
